From e80d720ef6f8fa244fd88e7e5d6115bda34abe47 Mon Sep 17 00:00:00 2001
From: Lei Maohui <leimaohui@fujitsu.com>
Date: Wed, 16 Aug 2023 08:59:56 +0000
Subject: [PATCH] Fix build error of jni.cpp:
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

| ../../../../../src/share/native/com/sun/java/util/jar/pack/jni.cpp:224:12: error: cannot convert ‘bool’ to ‘jobject’ {aka ‘_jobject*’} in return

Signed-off-by: Lei Maohui <leimaohui@fujitsu.com>
---
 .../jdk/src/share/native/com/sun/java/util/jar/pack/jni.cpp     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/openjdk/jdk/src/share/native/com/sun/java/util/jar/pack/jni.cpp b/openjdk/jdk/src/share/native/com/sun/java/util/jar/pack/jni.cpp
index e77e99f..3575425 100644
--- a/openjdk/jdk/src/share/native/com/sun/java/util/jar/pack/jni.cpp
+++ b/openjdk/jdk/src/share/native/com/sun/java/util/jar/pack/jni.cpp
@@ -221,7 +221,7 @@ Java_com_sun_java_util_jar_pack_NativeUnpack_getUnusedInput(JNIEnv *env, jobject
 
   if (uPtr->aborting()) {
     THROW_IOE(uPtr->get_abort_message());
-    return false;
+    return 0;
   }
 
   // We have fetched all the files.
-- 
2.34.1

